flocker/common/_interface.py

```diff
@@ -54,3 +54,27 @@ def class_decorator(cls):
             setattr(cls, name, method_decorator(name, *args, **kwargs))
         return cls
     return class_decorator
+
+
+def provides(interface):
+    """
+    Create an invariant that asserts that the given value provides the given
+    interface.
+
+    :param InterfaceClass interface: The interface to check for.
+
+    :return: A function that takes an object and returns a tuple of `bool` and
+        a error message.
+    """
+    interface_name = interface.__name__
+
+    def invariant(value):
+        if interface.providedBy(value):
```

Comment:
I don't want to pay the cost of generating the representation if it isn't needed.